tests/CMakeLists.txt

```diff
@@ -1,3 +1,6 @@
+if (NOT BUILD_TESTING)
+  return()
+endif()
```

Comment:
I think this should be impossible to reach, because we only include the `tests` directory in the top-level `CMakeLists.txt` if `BUILD_TESTING=ON`. Instead, I think maybe a `if (BUILD_UNITTESTS)` clause should be put around the `unittests` directory below?

Reply:
yes, good catch, thanks!